test/e2e_node/memory_manager_metrics_test.go

```diff
@@ -101,23 +108,28 @@ var _ = SIGDescribe("Memory Manager Metrics", framework.WithSerial(), feature.Me
 				[]memoryManagerCtnAttributes{
 					{
 						ctnName: "memmngrcnt",
-						cpus:    "100m",
+						cpus:    "1",
 						memory:  "1000Gi"},
-				}))
+				}),
+			)
+
+			printAllPodsOnNode(ctx, f.ClientSet, framework.TestContext.NodeName)
 
 			// we updated the kubelet config in BeforeEach, so we can assume we start fresh.
 			// being [Serial], we can also assume noone else but us is running pods.
 			ginkgo.By("Checking the memorymanager metrics right after the kubelet restart, with pod failed to admit")
 
 			matchResourceMetrics := gstruct.MatchKeys(gstruct.IgnoreExtras, gstruct.Keys{
 				"kubelet_memory_manager_pinning_requests_total": gstruct.MatchAllElements(nodeID, gstruct.Elements{
-					"": timelessSample(1),
+					"": memoryTimelessSample(1),
 				}),
 				"kubelet_memory_manager_pinning_errors_total": gstruct.MatchAllElements(nodeID, gstruct.Elements{
-					"": timelessSample(1),
+					"": memoryTimelessSample(1),
```

Comment:
We do indeed have leaked pods in some (many?) runs: https://github.com/kubernetes/kubernetes/pull/123468#issuecomment-1977977609

I can't pinpoint yet why we have unexpected pods, but being the case the tests run periodically/on-demand (vs against each PR), we will always lagging behind and we will always need to chase other e2e tests to make sure they do the cleanup properly. Since we are already severely undercapacity I don't think this would help.
I do agree the current check is too relaxed (but OTOH I also believe the former was too strict).
A better check would be compare total and errors, and ensure that in our testbed total > errors, I'll be working on that.

Reply:
> we will always lagging behind and we will always need to chase other e2e tests to make sure they do the cleanup properly

Since the tests run in serial fashion I don't think it would be too complex to delete all running guaranteed QoS pods before we start.
We should list all pods, filter by QoS and delete shouldn't be too expensive IMO

Reply:
true, but still this is a workaround, which makes the ecosystem slightly worse, despite good intention. The proper fix would be to fix the leaking tests instead of piling workarounds.

Reply:
I see, does the name of the rebellious pod shade some light about his origin?

Reply:
Yes, it's `test/e2e_node/container_manager_test.go`. I'm against adding checks in _this_ test because failing the preconditions will make _this_ test red, kinda hiding the misbehaving part.

Reply:
I agree. And I presume cleaning the `test/e2e_node/container_manager_test.go` properly is an option but it might bite us again with another careless cleanup in the future